modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaCheck.java

```diff
@@ -97,6 +114,43 @@ private Map<String, String> _getMap(String fileName) throws Exception {
 		return map;
 	}
 
+	private String _replaceImpl(
+		String content, String className, String newClassName) {
+
+		String[] lines = content.split(StringPool.NEW_LINE);
```

Comment:
We are splitting the file contents into lines and performing a linear scan every time we find a class name that has changed. This is somewhat inefficient. Can you try using something like `StringUtil.replace(content, new String[] {old1, old2...}, new String[] {new1, new2})`?